performance-tests/src/main/java/io/kroxylicious/benchmarks/InvokerDispatchBenchmark.java

```diff
@@ -0,0 +1,188 @@
+        String invoker;
+
+        private RequestHeaderData requestHeaders;
+        private KrpcFilterContext filterContext;
+        private Map.Entry<ApiKeys, ApiMessage>[] apiMessages;
+
+        @SuppressWarnings("unchecked")
+        @Setup
+        public void init() {
+            Invoker invokerType = Invoker.valueOf(invoker);
+            invokers = new FilterInvoker[]{
+                    invokerType.invokerWith(new FourInterfaceFilter0()),
+                    invokerType.invokerWith(new FourInterfaceFilter1()),
+                    invokerType.invokerWith(new FourInterfaceFilter2()),
+                    invokerType.invokerWith(new FourInterfaceFilter3()),
+                    invokerType.invokerWith(new FourInterfaceFilter0()),
+                    invokerType.invokerWith(new FourInterfaceFilter1()),
+                    invokerType.invokerWith(new FourInterfaceFilter2()),
+                    invokerType.invokerWith(new FourInterfaceFilter3())
+            };
+            final Map<ApiKeys, ApiMessage> messages = Map.of(ApiKeys.PRODUCE, new ProduceRequestData(), ApiKeys.API_VERSIONS, new ApiVersionsRequestData(), ApiKeys.FETCH,
+                    new FetchRequestData());
+            apiMessages = messages.entrySet().toArray(new Map.Entry[0]); // Avoids iterator.next showing up in the benchmarks
+            requestHeaders = new RequestHeaderData();
+            filterContext = new StubFilterContext();
+            keys = messages.keySet().toArray(new ApiKeys[0]);
+        }
+    }
+
+    @Benchmark
+    public void testDispatchToShouldHandle(BenchState state, Blackhole blackhole) {
+        invokeShouldHandle(blackhole, state.invokers, state.keys);
+    }
+
+    @Benchmark
+    public void testDispatchToHandleRequest(BenchState state) {
+        invokeHandleRequest(state.invokers, state.apiMessages, state.requestHeaders, state.filterContext);
+    }
+
+    @Benchmark
+    @Threads(4)
+    public void test4ThreadsDispatchToShouldHandle(BenchState state, Blackhole blackhole) {
+        invokeShouldHandle(blackhole, state.invokers, state.keys);
+    }
+
+    @Benchmark
+    @Threads(4)
+    public void test4ThreadsDispatchToHandleRequest(BenchState state) {
+        invokeHandleRequest(state.invokers, state.apiMessages, state.requestHeaders, state.filterContext);
+    }
+
+    private static void invokeShouldHandle(Blackhole blackhole, FilterInvoker[] filters, ApiKeys[] apiKeys) {
+        for (ApiKeys apiKey : apiKeys) {
+            final short apiVersion = apiKey.latestVersion();
+            for (FilterInvoker invoker : filters) {
+                blackhole.consume(invoker.shouldHandleRequest(apiKey, apiVersion));
+                blackhole.consume(invoker.shouldHandleResponse(apiKey, apiVersion));
+            }
+        }
+    }
+
+    private static void invokeHandleRequest(FilterInvoker[] filters, Map.Entry<ApiKeys, ApiMessage>[] apiMessages, RequestHeaderData requestHeaders,
```

Comment:
We could introduce a copy which calls `onResponse` as well though I'm not clear that would add any additional value.